src/eth1spec/eth_types.py

```diff
@@ -0,0 +1,127 @@
+"""
+Ethereum Types
+^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Types re-used throughout the specification, which are specific to Ethereum.
+"""
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from .base_types import U256, Bytes, Bytes8, Bytes20, Bytes32, Bytes64, Uint
+
+Address = Bytes20
+Root = Bytes
+Hash32 = Bytes32
+Hash64 = Bytes64
+
+Storage = Dict[Bytes32, U256]
+Bloom = Bytes32
+
+TX_BASE_COST = 21000
+TX_DATA_COST_PER_NON_ZERO = 68
+TX_DATA_COST_PER_ZERO = 4
+
+
+@dataclass
+class Transaction:
+    """
+    Atomic operation performed on the block chain.
+    """
+
+    nonce: U256
+    gas_price: U256
+    gas: U256
+    to: Optional[Address]
+    value: U256
+    data: Bytes
+    v: U256
+    r: U256
+    s: U256
+
+
+@dataclass
+class Account:
+    """
+    State associated with an address.
+    """
+
+    nonce: Uint
+    balance: Uint
+    code: bytes
+    storage: Storage
+
+
+EMPTY_ACCOUNT = Account(
+    nonce=Uint(0),
+    balance=Uint(0),
+    code=bytearray(),
+    storage={},
+)
+
+
+@dataclass
+class Header:
+    """
+    Header portion of a block on the chain.
+    """
+
+    parent: Hash32
+    ommers: Hash32
+    coinbase: Address
+    state_root: Root
+    transactions_root: Root
```

Comment:
`Root` type appears to allow arbitrary length values (since it is an alias for `Bytes`).  Should this and the others be `Hash32`?